src/analysis/normed/group/pointwise.lean

```diff
@@ -55,6 +72,18 @@ eq_of_forall_le_iff $ λ r, by simp_rw [le_inf_edist, ←image_inv, ball_image_i
 lemma inf_edist_inv_inv (x : E) (s : set E) : inf_edist x⁻¹ s⁻¹ = inf_edist x s :=
 by rw [inf_edist_inv, inv_inv]
 
+@[to_additive] lemma ediam_mul_le (x y : set E) :
+  emetric.diam (x * y) ≤ emetric.diam x + emetric.diam y :=
```

Comment:
Again, I think that we should add a lemma about `set.image2` under a map which is Lipschitz continuous in both variables, possibly with different Lipschitz constants.

Reply:
I've added that lemma but I don't see how to use it here.

Reply:
Yeah, your lemma crucially uses the fact that there's a lot of collisions between elements of `x` and `y` under multiplication (namely `x * y = z` has a lot of solutions for different `x` and `y`). It's hard/(impossible?) to capture that with `set.image2`.

Reply:
@YaelDillies: see my update, it was easy, we were thinking about the diam in normed _additive_ groups where `*` does not behave, but this lemma is about weird normed _multiplicative_ groups.